rebar.config.script

```diff
@@ -13,7 +13,10 @@ NoHaveMapsOpts = try maps:size(maps:new()) of
     %% needed with rebar for windows/cygwin
     "sh build/mk_version_hrl"
     "    < include/gpb_version.hrl.in"
-    "    > include/gpb_version.hrl "}
+    "    > include/gpb_version.hrl "},
+   {compile,
+    %% way of invoking shell script: see above
+    "sh build/compile_descriptor"}
```

Comment:
I don't think moving the compile_descriptor from a post-hook to a pre-hook is good.

It is a two-stage process to build gpb (or three-stage if you include building the `gpb_version.hrl`):
1. First build the compiler
2. Then use the compiler to build an encoder for `.proto` files. This is the descriptor. It can be used to add a protobuf-encoded description of the proto file(s) into generated `.erl` files, if the `descriptor' (`-descr') option is specified.

On this branch, `make clean` followed by `rebar compile` now leads to:
```
% make clean
rm -f include/gpb_version.hrl ...
...
% rebar compile
==> gpb (compile)
Compiling descriptor.proto...
{"init terminating in do_boot",{undef,[{gpb_compile,c,[['descriptor.proto']],[]},{init,start_it,1,[]},{init,start_em,1,[]}]}}
```
Maybe there's some other way to make this work with both rebar2 and rebar3? (not knowing what the problem with rebar3 was)